### PR TITLE
Feat/display card list

### PR DESCRIPTION
**Pull request by CharlesSquirel**

### CODE REVIEW

**Review by Sumick — CHANGES_REQUESTED** [review]
There are unrelated files in this PR. Please, pull main first, and leave only task relevant files.

**Comment by CharlesSquirel** [comment]
Is there any chance for checking my test refactoring, even if it is not related to the task? Or it will be reviewed at the QA stage?

File: src/App.tsx [hunk 1]
```diff
@@ -25,12 +25,17 @@ function App() {
   const [flashCards, setFlashCards] = useState([]);
   const scrollContainerRef = useRef(null);
 
+  const getAllCards = async () => {
+    const cards = await getCards();
+    setFlashCards(cards);
```
Comment (Sumick) [inline src/App.tsx]: **nitpick:** This function is missing error handling

File: src/components/Card/CardEdit.tsx [hunk 2]
```diff
@@ -19,13 +19,17 @@ const CardEdit: React.FC<ICardEdit> = ({ handleSwitchEdit, type, card }) => {
   const handleSave = async () => {
     await updateCard(card, textToEdit);
     handleSwitchEdit();
-    await getCards(setFlashCards);
+    if (getAllCards) {
+      await getAllCards();
```
Comment (Sumick) [inline src/components/Card/CardEdit.tsx]: **nitpick:** You'd have to add these `if` all the time adding complexity to your codebase - let's fix it, so you don't have to do this.

File: src/components/services/api/api.ts [hunk 3]
```diff
@@ -26,10 +26,17 @@ export const addCard = async (card: CardTextsProps) => {
...
-  const data = await res.json();
-  settingFunction(data ? data : null);
+export const getCards = async () => {
+  try {
+    const res = await fetch(url);
+    const data = await res.json();
+    if (!res.ok) {
+      throw new Error("Network response was not ok");
+    }
+    return data;
```
Comment (Sumick) [inline src/components/services/api/api.ts]: **nitpick:** This function is missing return type.

File: src/components/services/types/types.ts [hunk 4]
```diff
@@ -70,6 +70,7 @@ export interface ContextProps {
   flashCards: IFlashcard[];
   setFlashCards: React.Dispatch<React.SetStateAction<any>>;
   scrollContainerRef: React.RefObject<HTMLDivElement> | null;
+  getAllCards?: () => {};
```
Comment (Sumick) [inline src/components/services/types/types.ts]: **nitpick:** Why optional?

File: src/tests/App.test.tsx [hunk 5]
```diff
@@ -49,113 +45,115 @@ const flaschCardTest = {
...
+    expect(saveBtn).toBeDisabled();
+    fireEvent.change(editInput, { target: { value: newValue } });
+    expect(editInput.value).toEqual(newValue);
+    fireEvent.click(saveBtn);
+    expect(editCard).not.toBeInTheDocument();
+  });
+  const frontCardInput = screen.getAllByTestId("card-text")[0] as HTMLTextAreaElement;
+  expect(frontCardInput).toBeInTheDocument();
+  // await waitFor(() => {
+  //   expect(frontCardInput).toEqual(newValue)
```
Comment (Sumick) [inline src/tests/App.test.tsx]: **nitpick:** comment to remove

**Review by Sumick — APPROVED** [review]
Tests look nice - good job! I've found few things we should improve before merging

**Pull request merged**